maintenance/zero_downtime_migration.rst

```diff
@@ -0,0 +1,167 @@
+.. _zero_downtime_migration:
+
+Zero-downtime Migration
+===============================
+
+Firely Server v6 introduces an enhanced MongoDB database migration process. In contrast to the previous migration method that required system downtime, this new approach enables zero-downtime migration. This allows for continuous operation of your existing Firely Server instance while the data migration to the new schema proceeds.
+
+.. note::
+   This migration process applies to the *vonkdata* database. 
+   
+   If you also use MongoDB as the **admin** database, you will need to initialize a new *vonkadmin* database from scratch by specifying a different connection string for the admin database than your previous installation.
+
+.. warning::
+   Please note that any snapshot produced by the bulk data export functionality will not be available after the migration.
+
+.. important::
+   Bulk data export data is excluded from the migration process.
+
+Architecture Overview
+---------------------
+
+The following diagram illustrates the architecture and components involved in the migration process:
+
+.. image:: migration-diagram.svg
+   :alt: Migration Diagram
+
+The architecture consists of:
+
+* An existing database containing the source data
+* The current Firely Server instance maintaining normal operations during migration
+* A new target database instance initialized empty
+* A migration tool operating asynchronously to transform and transfer data
+* A new Firely Server instance configured for the target database
+* A reverse proxy managing request routing between instances
+
+.. note::
+   During the migration period, both database instances operate concurrently.
+
+Prerequisites
+-------------
+
+Before starting the migration, ensure you have:
+
+* Firely Server Integration (FSI) tool downloaded
+* Sufficient disk space for both databases during migration
+* License file with the token ``http://fire.ly/vonk/plugins/bulk-data-import/migration``
+* Access credentials for both source and target MongoDB instances
+* MongoDB Shell (mongosh) installed for sharding configuration
+
+Implementation Process
+----------------------
+
+1. Determine the target database deployment location and make a note of the connection string for reference in the next steps.
+
+   If you use MongoDB as the **admin** database, you will need to initialize a new *vonkadmin* database from scratch by specifying a different connection string for the admin database than your previous installation. If you imported custom conformance resources, make sure to include them in your nw installation. See :ref:`conformance_import` for more information. 
+
+2. For implementations requiring sharding, execute the following configuration steps:
+
+   - Ensure your MongoDB installation supports sharding. Refer to the MongoDB documentation if uncertain.
+   - Provision the schema using the following command:
+
+      .. code-block:: bash
+
+         COLLECTION_NAME=vonkentries
```

Comment:
Do all scripts work in both bash and Powershell? If not, for which are they written?

Reply:
I have added separate scripts for bash and powershell.